.github/workflows/lint.yml

```diff
@@ -7,21 +10,31 @@ on:
       - develop
 
 jobs:
+  # Running pre-commit in CI. See semgrep/.pre-commit-config.yaml for
+  # our pre-commit configuration.
   pre-commit:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        #TODO: why do we need this ref:? doesn't checkout@v3 grabs the right branch by default?
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+      # We grab those submodules below because they are the one needed by 'mypy' which
+      # runs as part of pre-commit to check our Python code.
+      # alt: we could also use 'submodules: recursive' instead, but that would be slower
       - name: Fetch semgrep-cli submodules
         run: git submodule update --init --recursive --recommend-shallow cli/src/semgrep/lang cli/src/semgrep/semgrep_interfaces
+      #TODO: why this is needed? pre-commit/action@v3 requires this setup-python to run?
+      # what is this cache-dependency-path?
```

Comment:
This caches the pip installed dependencies that pre-commit uses. Otherwise it will need to reinstall everytime

Reply:
And yes pre-commit is a python script :D https://github.com/pre-commit/pre-commit

Reply:
so each time you use the pre-commit/action, you also need to setup python before? The github action don't have a way to pull automatic dependencies?